### ISSUE TITLE
Backend services unaware of request scheme

### ISSUE THREAD

**Issue by easkay-castlight** [haproxytech/kubernetes-ingress#124]
Hi folks,

I have a webapp running in kubernetes, with HAProxy as the ingress controller. The webapp uses the incoming request (including headers) to determine what scheme to use when generating URLs. I've noticed that when accessing the app via the ingress controller, some links are HTTP instead of HTTPS, meaning that the app is unaware of what scheme was used for the original request.

In the past, I have solved this issue by setting `X-Forwarded-Proto` to `https` to inform the webapp that the original request was HTTPS and therefore the generated links are HTTPS as well.

Is this something that the ingress controller could support? I believe a relatively simple set-header in the frontend would do the trick:
```
http-request set-header X-Forwarded-Proto https if { ssl_fc }
```
(adapted from https://stackoverflow.com/questions/51928504/x-forwarded-proto-https-in-frontend-or-backend-haproxy).

I'd be happy to raise a PR to add the functionality if you'd like.

Thanks for your consideration.

**Comment by oktalz** [comment]
hi @easkay-castlight 
this seem like a good option to have

PR is definitely welcomed here :)

**Comment by easkay-castlight** [comment]
Cool, thanks @oktalz. I'll try and have a look soon.

**Comment by easkay** [comment]
@oktalz I have a couple of ideas about where/how to implement this, but I had a couple of questions:

* Should this behaviour have an option to control it? E.g. configmap/annotations/CLI args
* If not, where would be the best place to add the HTTPRequestsRule? I haven't yet fully studied the code and as yet don't know what happens at what point, so some advice would be greatly appreciated

If it's better to discuss these things in a PR, then I'm happy to put a draft in.

**Comment by easkay-castlight** [comment]
This was fixed in #146.
